packages/gatsby-theme-bodiless/src/dist/GatsbyMobxStore.ts

```diff
@@ -290,4 +150,25 @@ export default class GatsbyMobxStore {
       this.setItem(key, new Item(this, key, value, event));
     }
   };
+
+  getChildrenNodes = (keyPath: string[]) => {
+    const key = keyPath.join(nodeChildDelimiter);
+    const children = Array.from(this.store)
+      .filter(item => item[0].indexOf(key) === 0 && item[0] !== key);
+    return children;
+  };
+
+  deleteItem = (key: string) => {
+    const item = this.store.get(key);
+    return item && item.delete();
+  };
+
+  deleteNode = (keyPath: string[]) => {
+    const children = this.getChildrenNodes(keyPath);
+    children.forEach(child => {
```

Comment:
Would it be better to implement this logic on the backend, rather than sending a slew or delete requests?  I know ti's less RESTful - but should we be concerned about a large number of near-simultaneous http requests?

Reply:
@wodenx, the concern is whether we have a large number delete requests. we have delete requests triggered by list component and by flexbox. In the list case, we can get a lot of requests when we delete an item that has a lot of children. not sure it will happen very often.
in the flexbox case, we can get a lot of requests when we delete an item with a lot of children nodes.  not sure it will happen very often either.
if amount of requests becomes a problem for us, then I believe we can consider consolidating the requests on backend client side and/or change http to websockets